lib/net.c

```diff
@@ -154,120 +154,196 @@ ethertype_filter_add(struct gatekeeper_if *iface, uint16_t ether_type,
 }
 
 /*
- * @dst_ip, @src_port, @src_port_mask, @dst_port, and @dst_port_mask
- * must be in big endian.
- *
- * By specifying the tuple (proto, src_port, dst_port) (and masks),
- * it can filter both IPv4 and IPv6 addresses.
+ * @dst_ip_be, @src_port_be, @src_port_mask_be, @dst_port_be, and
+ * @dst_port_mask_be must be in big endian.
  */
-int
-ntuple_filter_add(struct gatekeeper_if *iface, uint32_t dst_ip,
-	uint16_t src_port, uint16_t src_port_mask,
-	uint16_t dst_port, uint16_t dst_port_mask,
-	uint8_t proto, uint16_t queue_id,
-	int ipv4_configured, int ipv6_configured)
+static int
+ipv4_ntuple_filter_add(struct gatekeeper_if *iface, rte_be32_t dst_ip_be,
+	rte_be16_t src_port_be, rte_be16_t src_port_mask_be,
+	rte_be16_t dst_port_be, rte_be16_t dst_port_mask_be,
```

Comment:
Given that we already express that `@dst_ip_be, @src_port_be, @src_port_mask_be, @dst_port_be, and @dst_port_mask_be` are big endian in their types and names, it's safe to remove the comment above.